### ISSUE TITLE
Request confirmation before deleting comment

### ISSUE THREAD

**Issue by ffont** [MTG/freesound#1204]
Currently authors of a sound can delete the comments that other users make to that sound. Some users reported that it is easy to accidentally delete a comment because there is no confirmation step. We should open a simple "are you sure?" dialog before deleting. With jQuery we can do styled dialogs easily, otherwise native dialog would be fine as well. See https://stackoverflow.com/questions/12531324/javascript-or-jquery-are-you-sure-dialog-for-a-link/12531454#12531454

**Comment by alastair** [comment]
We have this workflow in many places, we should consider if we want to standardise the process: https://github.com/MTG/freesound/pull/1191/commits/4b3765fc7d0e1b60726597f0eeb7590fc5ba9ec0
Personally I'm not a huge fan of using js directly for this operation.

**Comment by ffont** [comment]
I thought of JS for simplicity in the code. We don't use dialogs often in FS, but in the new interface we'll use some modals and I think these kind of things would be very easy to implement with modal + JS. What are the reasons not to use JS here other than people not running JS? Could we do something like "if user does not run JS, then there is no delete confirmation". Although I'd like to know how may users don't enable JS in their browsers...

**Comment by alastair** [comment]
While we're looking at this ticket let's also try and find all other places where we have a delete link (and optional confirmation page) workflow. Ideally we should make all of these work the same way, regardless of what that way is.

**Comment by alastair** [comment]
also #1288

**Comment by ffont** [comment]
A confirmation modal has been added in the new UI, but we should double check if the request is GET or POST and make it POST.

**Comment by alastair** [comment]
This modal uses GET

**Comment by kashyap-harshit** [comment]
Is this issue resolved? Because I am getting a confirmation dialogue whenever I try to delete a comment on the sound page.

**Comment by ffont** [comment]
Yes the confirmation modal is in place, but is uses an http GET method instead of POST to make the deletion request. So what is missing to do is implement comment deletion as POST request instead of GET.

**Comment by kashyap-harshit** [comment]
<img width="1601" height="485" alt="Image" src="https://github.com/user-attachments/assets/03bd8bf2-61de-457f-8844-02c1c3ffad01" />

Correct me if I am wrong, but the frontend is sending a POST method I believe to delete a comment.
But I was going through the backend (comments/views.py) the delete function is not method safe that is even with a GET method the message can be deleted. Now would you like me to fix that? 
I'd add Django's @require_POST decorator to the delete view so it rejects GET requests.
